xarray/core/computation.py

```diff
@@ -1953,15 +1962,21 @@ def _ensure_numeric(data: Dataset | DataArray) -> Dataset | DataArray:
     """
     from .dataset import Dataset
 
+    def _cfoffset(x: DataArray) -> Any:
+        scalar = x.compute().data[0]
+        if not is_scalar(scalar):
+            # we do not get a scalar back on dask == 2021.04.1
+            scalar = scalar.item()
+        return type(scalar)(1970, 1, 1)
+
     def to_floatable(x: DataArray) -> DataArray:
-        if x.dtype.kind == "M":
-            # datetimes
+        if x.dtype.kind in "MO":
+            # datetimes (CFIndexes are object type)
+            offset = (
+                np.datetime64("1970-01-01") if x.dtype.kind == "M" else _cfoffset(x)
```

Comment:
can we get a test with a non-standard calendar / something very far from 1970 (like, `480-01-01`)? I suspect that will cause the fix to fail because this hard-codes the offset to `1970-01-01` (not sure, though, I might misunderstand the code).

Reply:
This was added to copy the behavior of `polyfit` . But I agree that such a test can be useful. Additionally we should add an integration test for these two functions.

Reply:
That offset choice is in `main` too: https://github.com/pydata/xarray/blob/9b4d0b29c319f4b68f89328b1bf558711f339504/xarray/core/missing.py#L275-L280